File: View/SaveView.cs
using Isocline.Controller;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Isocline.View
{
    internal class SaveView
    {
        private PictureBox picture;
        private SaveController saveController;
        public SaveView(PictureBox picture, SaveController saveController)
        {
            this.picture = picture;
            this.saveController = saveController;
        }

        public void SaveButtonClick(object sender, EventArgs e)
        {
            if (picture.Image != null)
            {
                SaveFileDialog sfd = new SaveFileDialog();
                sfd.Title = "Зберегти зображення як...";
                sfd.OverwritePrompt = true;
                sfd.CheckPathExists = true;
                sfd.Filter = "Image Files(*.BMP)|*.BMP|Image Files(*.JPG)|*.JPG|Image Files(*.PNG)|*.PNG|All files(*.*)|*.*";
                sfd.ShowHelp = true;

                if (sfd.ShowDialog() == DialogResult.OK)
                {
                    try
                    {
                        saveController.SaveImage(sfd.FileName);
                    }
                    catch
                    {
                        MessageBox.Show("Неможливо зберегти зображення", "Помилка", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }
                }

            }
        }
    }
}

File: View/SaveView.vb
Imports Isocline.Controller
Imports System
Imports System.Windows.Forms

Namespace Isocline.View
    Friend Class SaveView
        Private picture As PictureBox
        Private saveController As SaveController
        Public Sub New(ByVal picture As PictureBox, ByVal saveController As SaveController)
            Me.picture = picture
            Me.saveController = saveController
        End Sub

        Public Sub SaveButtonClick(ByVal sender As Object, ByVal e As EventArgs)
            If picture.Image IsNot Nothing Then
                Dim sfd As SaveFileDialog = New SaveFileDialog()
                sfd.Title = "Зберегти зображення як..."
                sfd.OverwritePrompt = True
                sfd.CheckPathExists = True
                sfd.Filter = "Image Files(*.BMP)|*.BMP|Image Files(*.JPG)|*.JPG|Image Files(*.PNG)|*.PNG|All files(*.*)|*.*"
                sfd.ShowHelp = True

                If sfd.ShowDialog() = DialogResult.OK Then
                    Try
                        saveController.SaveImage(sfd.FileName)
                    Catch
                        MessageBox.Show("Неможливо зберегти зображення", "Помилка", MessageBoxButtons.OK, MessageBoxIcon.Error)
                    End Try
                End If

            End If
        End Sub
    End Class
End Namespace
